odoo/orm/fields_properties.py

```diff
@@ -223,7 +225,9 @@ def _get_res_ids_per_model(self, env, values_list):
 
                 if type_ == 'many2one':
                     default = [default] if default else []
-                    property_value = [property_value] if isinstance(property_value, int) else []
+                    property_value = [property_value] if isinstance(property_value, int) else [property_value.id] if isinstance(property_value, BaseModel) else []
```

Comment:
An empty recordset will fill in `False` in `res_ids_per_model`.
```suggestion
                    property_value = [property_value] if isinstance(property_value, int) else property_value.ids if isinstance(property_value, BaseModel) else []
```